core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -3200,8 +3209,41 @@ private boolean o3Commit(long lag) {
             // Safe check of the sort. No known way to reproduce
             assert o3TimestampMin <= o3TimestampMax;
 
-            if (lag > 0) {
-                long lagThresholdTimestamp = o3TimestampMax - lag;
+            if (o3MaxLag > 0) {
+                long lagError = 0;
+                if (getMaxTimestamp() != Long.MIN_VALUE) {
+
+                    // When table already has data we can calculate the overlap of the newly added
```

Comment:
Sometimes all rows will be in lag. Would be good to detect this before sorting the data and return early.

Reply:
we can address this in another PR, this one is creeping up nicely already